apiserver/facades/client/bundle/bundle.go

```diff
@@ -150,11 +160,149 @@ func (b *BundleAPI) GetChanges(args params.BundleChangesParams) (params.BundleCh
 	return results, nil
 }
 
+var fail = func(failErr error) (params.StringResult, error) {
+	return params.StringResult{}, common.ServerError(failErr)
+}
+
 // ExportBundle exports the current model configuration as bundle.
 func (b *BundleAPI) ExportBundle() (params.StringResult, error) {
-	return params.StringResult{}, nil
+	if err := b.checkCanRead(); err != nil {
+		return fail(err)
+	}
+
+	var exportConfig state.ExportConfig
+
+	b.backend.SetExportconfig(exportConfig)
```

Comment:
Why do u need to it this way? What we discussed this morning would have looked something like https://pastebin.ubuntu.com/p/skHsmffKgb/

This way, you next call becomes

```
b.backend.ExportPartial()
```